R/create_custom_check.R

```diff
@@ -0,0 +1,113 @@
+#' @param hub_path Character string. Path to the hub directory. Default is the
+#' current working directory.
+#' @param r_dir Character string. Path (relative to `hub_path`) to the directory
+#' the custom check function file will be written to. Default is `src/validations/R`
+#' which is the recommended directory for storing custom check functions.
+#' @param error Logical. If `TRUE`, the custom check function will return a
+#' `<error/check_error>` class object in the case of a failed check instead of the
+#' default `<error/check_failure>`.
+#' @param conditional Logical. If `TRUE`, the custom check function template will
+#' include a block of code to check a condition before running the check. This is useful
+#' when a check may need to be skipped based on a condition.
+#' @param error_object Logical. If `TRUE`, the custom check function template will
+#' include an error object that can be used to store additional information about the
+#' properties of the object being checked that caused check failure. For example,
+#' it could store the index of rows in a `tbl` that caused a check failure.
+#' @param extra_args Logical. If `TRUE`, the custom check function template will
+#' include an `extra_arg` template argument and template block of code to check
+#' the input arguments of the custom check function.
+#' @param overwrite Logical. If `TRUE`, the function will overwrite an existing
+#'
+#' @return Invisible `TRUE` if the custom check function file is created successfully.
+#' @details
+#' See the article on [writing custom check functions](
+#' https://hubverse-org.github.io/hubValidations/articles/writing-custom-fns.html)
+#' for more.
+#' @export
+#'
+#' @examples
+#' withr::with_tempdir({
+#'   # Create the custom check file with default settings.
+#'   create_custom_check("check_default")
+#'   cat(readLines("src/validations/R/check_default.R"), sep = "\n")
+#'
+#'  # Create fully featured custom check file.
+#'   create_custom_check("check_full",
+#'     error = TRUE, conditional = TRUE,
+#'     error_object = TRUE, extra_args = TRUE
+#'   )
+#'   cat(readLines("src/validations/R/check_full.R"), sep = "\n")
+#' })
+create_custom_check <- function(name, hub_path = ".",
+                                r_dir = "src/validations/R",
+                                error = FALSE, conditional = FALSE,
+                                error_object = FALSE, extra_args = FALSE,
+                                overwrite = FALSE) {
+  checkmate::assert_character(name, len = 1L)
+  checkmate::assert_scalar(hub_path)
+  checkmate::assert_scalar(r_dir)
+  checkmate::assert_logical(error, len = 1L)
+  checkmate::assert_logical(conditional, len = 1L)
+  checkmate::assert_logical(error_object, len = 1L)
+  checkmate::assert_logical(extra_args, len = 1L)
+  checkmate::assert_logical(overwrite, len = 1L)
+
+  # If hub_path not current working directory, prefix path to R dir with hub_path.
```

Comment:
Should add a check that the `hub_path` exists and is a hub.

Reply:
I've added a check to ensure a `hub_path` directory exists but don't want to be too prescriptive about the contents of `hub_path`. Not much is gained as a fully functional hub is not required for `create_custom_check()` to work correctly, it would make it difficult for folks to just create templates in a throwaway folder to play around with (including our own tests!) and there is no hard and fast rule for identifying a hub. We usually check for the part of the hub that is relevant to a given functionality which is not obvious here.